### PR TITLE
CPDRP-260 Add API for Data Studio

### PR DESCRIPTION
**Pull request by tonyheadford**
### Context
[Jira](https://dfedigital.atlassian.net/browse/CPDRP-260)
Create API endpoint that Data Studio can use to fetch analytics data for school rollout
This is the first step in getting an automatic feed of data into Data Studio.

### Changes proposed in this pull request
Add endpoint to expose school rollout data used by Data Studio
Add `DataStudioApiToken` as a private auth token

### Guidance to review
You need an API key.
The endpoint `/api/v1/data-studio/school-rollout` should return ~12Mb of school related data

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### CODE REVIEW

File: app/controllers/api/v1/data_studio/school_rollout_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          render json: ::DataStudio::SchoolRolloutSerializer.new(paginate(rollout_data))
+            .serializable_hash.to_json
+        end
+
+      private
+
+        def rollout_data
+          School
+            .eligible
+            .joins(:nomination_emails)
```
Comment (ethanmills) [inline app/controllers/api/v1/data_studio/school_rollout_controller.rb:20]: Once we start updating schools from GIAS, we will have eligible schools who have not been sent a nomination email

File: app/controllers/api/v1/data_studio/school_rollout_controller.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module DataStudio
+      class SchoolRolloutController < Api::ApiController
```
Comment (ethanmills) [inline app/controllers/api/v1/data_studio/school_rollout_controller.rb:6]: I think we need to stop lead provider tokens from working with this endpoint. This includes details of schools who have opted out, which we don't want to share with them. I'm also not sure we're allowed to share which providers have recruited which schools, which we will be adding very soon

Comment (tonyheadford) [reply app/controllers/api/v1/data_studio/school_rollout_controller.rb:6]: Yes, was going to add the private endpoint code that @asmega wrote

File: app/controllers/api/v1/data_studio/school_rollout_controller.rb (hunk 1 above)
Comment (tonyheadford) [reply app/controllers/api/v1/data_studio/school_rollout_controller.rb:20]: Thanks, good to know so that needs to be a left outer join then

**Pull request closed without merging**